src/Ilios/CoreBundle/Service/Fetch.php

```diff
@@ -30,11 +32,34 @@ public function __construct(HttpClient $client, RequestFactory $requestFactory)
         $this->requestFactory = $requestFactory;
     }
 
-    public function get(string $url) : string
+    /**
+     * Get a file
+     *
+     * If passed a $file it will use that to check if the response would be a 304 Not Modified
+     * and if so just return the contents of $file to save bandwidth downloading it again
+     * 
+     * @param string $url
+     * @param \SplFileObject|null $file
+     * @return string
+     * @throws \Exception
+     */
+    public function get(string $url, \SplFileObject $file = null) : string
     {
         $request = $this->requestFactory->createRequest('GET', $url);
+
+        if ($file) {
+            $lastModifiedTime = $file->getMTime();
+            $lastModified = new DateTime();
+            $lastModified->setTimestamp($lastModifiedTime);
+            $request = $request->withHeader('if-modified-since', $lastModified->format('D, d M Y H:i:s T'));
+        }
+
         $response = $this->client->sendRequest($request);
-        $fileContents = $response->getBody()->getContents();
+        if ($response->getStatusCode() === Response::HTTP_NOT_MODIFIED) {
```

Comment:
this will blow up if $file is NULL but the response is "not modified". please fix.